plugins/intel-me/fu-intel-me-mca-device.c

```diff
@@ -0,0 +1,256 @@
+		return FALSE;
+	}
+	if (hdr_req.command != hdr_res.command) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "invalid response command, requested 0x%x and got 0x%x",
+			    (guint)hdr_req.command,
+			    (guint)hdr_res.command);
+		return FALSE;
+	}
+	if (!hdr_res.is_resp) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "invalid response group ID, not a response!");
+		return FALSE;
+	}
+	if (hdr_res.result != MKHI_STATUS_SUCCESS) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "failed to get result, got 0x%x",
+			    hdr_res.result);
+		return FALSE;
+	}
+
+	/* verify payload size */
+	if (!fu_memread_uint32_safe(buf_res,
+				    bufsz_res,
+				    sizeof(FuMkhiHeader),
+				    &datasz_res,
+				    G_LITTLE_ENDIAN,
+				    error))
+		return FALSE;
+	if (datasz_res != datasz_req) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "invalid response data size, requested 0x%x and got 0x%x",
+			    datasz_req,
+			    datasz_res);
+		return FALSE;
+	}
+
+	/* create checksum */
+	for (gsize i = 0; i < datasz_req; i++) {
+		guint8 tmp = 0;
+		if (!fu_memread_uint8_safe(buf_res,
+					   bufsz_res,
+					   sizeof(FuMkhiHeader) + sizeof(guint32) + i,
+					   &tmp,
+					   error))
+			return FALSE;
+		g_string_append_printf(checksum, "%02x", tmp);
+	}
+	fu_device_add_checksum(FU_DEVICE(self), checksum->str);
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_intel_me_mca_device_setup(FuDevice *device, GError **error)
+{
+	FuIntelMeMcaDevice *self = FU_INTEL_ME_MCA_DEVICE(device);
+
+	if (!fu_intel_me_mca_device_add_checksum(self, MCA_SECTION_ME, error)) {
+		g_prefix_error(error, "failed to get OEM Public Key Hash ME FW: ");
+		return FALSE;
+	}
+	if (!fu_intel_me_mca_device_add_checksum(self, MCA_SECTION_UEP, error)) {
+		g_prefix_error(error, "failed to get OEM Public Key Hash UEP: ");
+		return FALSE;
+	}
+	if (!fu_intel_me_mca_device_add_checksum(self, MCA_SECTION_FPF, error)) {
+		g_prefix_error(error, "failed to get OEM Public Key Hash FPF: ");
+		return FALSE;
+	}
+
+	/* add guid */
+	fu_device_add_parent_guid(device, "main-system-firmware");
+
+	/* success */
+	return TRUE;
+}
+
+static void
+fu_intel_me_mca_device_init(FuIntelMeMcaDevice *self)
+{
+	fu_device_set_logical_id(FU_DEVICE(self), "MCA");
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_INTERNAL);
+	fu_device_add_icon(FU_DEVICE(self), "computer");
+	fu_device_set_name(FU_DEVICE(self), "BootGuard Configuration");
+}
+
+static void
+fu_intel_me_mca_device_add_security_attrs(FuDevice *device, FuSecurityAttrs *attrs)
+{
+	FuIntelMeMcaDevice *self = FU_INTEL_ME_MCA_DEVICE(device);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	const gchar *leaked_sha256[] = {
+	    "WAITING-FOR-DATA-FROM-LENOVO0a3f3be5464e04bc3423e8648860a8c9d9b2",
+	    "WAITING-FOR-DATA-FROM-LENOVO6f2ef3296fc9eca86bff3539801e9e265b22",
```

Comment:
This information isn't available yet from the leak?

Reply:
I've added the keys which I've calculated.